### PR TITLE
Setup Zulip notification with test results

### PR DESCRIPTION
**Pull request by radtriste**
This will get all info from Jenkins to display job result and test reports if failure/unstable, also get console logs (either from a console.log artifact if existing or directly from the console log)

This is formatting for Zulip (aka Markdown)

Examples of messages can be found in https://kie.zulipchat.com/#narrow/stream/236603-kogito-ci/topic/.5Btest.5D.20Test.20display.20failed.20tests

### CODE REVIEW

**Comment by radtriste** [comment]
Once that is merged I will update Kogito jobs to use those methods.

**Comment by radtriste** [comment]
created https://issues.redhat.com/browse/BXMSPROD-1577 to include this in nightlies/PR jobs

**Comment by radtriste** [comment]
Set as draft as I still need to run a final test ;)

**Comment by radtriste** [comment]
All good ! :)

File: vars/util.groovy [hunk 1]
```diff
@@ -365,9 +371,9 @@ def retrieveFailedTests(String buildUrl = "${BUILD_URL}") {
     def testResults = retrieveTestResults(buildUrl)
 
     def failedTests = []
-    testResults.suites.each { testSuite ->
-        testSuite.cases.each { testCase ->
-            if (testCase.status != 'PASSED' && testCase.status != 'SKIPPED') {
+    testResults.suites?.each { testSuite ->
+        testSuite.cases?.each { testCase ->
+            if (testCase.status != 'PASSED' && testCase.status != 'SKIPPED' && testCase.status != 'FIXED') {
```
Comment (Ginxo) [inline vars/util.groovy:376]:
```suggestion
            if (!['PASSED', 'SKIPPED', 'FIXED'].contains(testCase.status)) {
```
This way we can add/remove any value without adding new condition

**Review by Ginxo — APPROVED** [review]
LGTM, just a simple proposal

**Review by Ginxo — APPROVED** [review]
LGTM

**Comment by radtriste** [comment]
jenkins retest this

**Pull request merged**